python/src/pipeline/pipeline.py

```diff
@@ -441,7 +440,9 @@ def __init__(self, *args, **kwargs):
     self.outputs = None
     self.backoff_seconds = _DEFAULT_BACKOFF_SECONDS
     self.backoff_factor = _DEFAULT_BACKOFF_FACTOR
-    self.max_attempts = _DEFAULT_MAX_ATTEMPTS
+    # Allow pipeline class to provide it's own default max attempts value.
+    self.max_attempts = (getattr(self, "MAX_ATTEMPTS", _DEFAULT_MAX_ATTEMPTS) or
+                        _DEFAULT_MAX_ATTEMPTS)
```

Comment:
The `or _DEFAULT_MAX_ATTEMPTS` could be justified to allow a subclass to programmatically revert to the default by setting `self.MAX_ATTEMPTS = None` e.g.:

```
>>> class Foo(object):
...   bar = None
...
>>> f = Foo()
>>> getattr(f, 'bar', 2) == None
True
>>> getattr(f, 'bar', 2) or 2
2
```

Removing the extra clause would require deleting the attribute, which doesn't seem to work:

```
>>> del f.bar
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Foo' object attribute 'bar' is read-only
```